simp_le.py

```diff
@@ -509,45 +417,196 @@ class OpenSSLIOPlugin(IOPlugin):  # pylint: disable=abstract-method
+        return self.save_to_file(key)
+
+
+@IOPlugin.register(path='key.der', typ=OpenSSL.crypto.FILETYPE_ASN1)
+@IOPlugin.register(path='key.pem', typ=OpenSSL.crypto.FILETYPE_PEM)
+class KeyFile(FileIOPlugin, OpenSSLIOPlugin):
+    """Private key file plugin."""
+
+    def persisted(self):
+        return self.Data(account_key=False, key=True, cert=False, chain=False)
+
+    def load_from_content(self, content):
+        return self.Data(
+            account_key=None,
+            key=self.load_key(content),
+            cert=None,
+            chain=None,
+        )
+
+    def save(self, data):
+        key = self.dump_key(data.key)
+        return self.save_to_file(key)
+
+
+@IOPlugin.register(path='cert.der', typ=OpenSSL.crypto.FILETYPE_ASN1)
+@IOPlugin.register(path='cert.pem', typ=OpenSSL.crypto.FILETYPE_PEM)
+class CertFile(FileIOPlugin, OpenSSLIOPlugin):
+    """Certificate file plugin."""
+
+    def persisted(self):
+        return self.Data(account_key=False, key=False, cert=True, chain=False)
+
+    def load_from_content(self, content):
+        return self.Data(
+            account_key=None,
+            key=None,
+            cert=self.load_cert(content),
+            chain=None,
+        )
+
+    def save(self, data):
+        cert = self.dump_cert(data.cert)
+        return self.save_to_file(cert)
+
 
+@IOPlugin.register(path='chain.pem', typ=OpenSSL.crypto.FILETYPE_PEM)
+class ChainFile(FileIOPlugin, OpenSSLIOPlugin):
+    """Certificate chain plugin."""
 
-def dump_pem_jwk(data):
-    """Dump JWK as PEM."""
-    return data.key.private_bytes(
-        encoding=serialization.Encoding.PEM,
-        format=serialization.PrivateFormat.TraditionalOpenSSL,
-        encryption_algorithm=serialization.NoEncryption(),
-    ).strip()
+    def persisted(self):
+        return self.Data(account_key=False, key=False, cert=False, chain=True)
+
+    def load_from_content(self, content):
+        pems = [pem for pem in self.split_pems(content)]
+        if not pems:
+            raise Error("No PEM encoded message was found on {0}, "
+                        "at least 1 was expected.".format(self.path))
+        return self.Data(
+            account_key=None,
+            key=None,
+            cert=None,
+            chain=[self.load_cert(cert) for cert in pems[0:]],
+        )
+
+    def save(self, data):
+        pems = [self.dump_cert(cert) for cert in data.chain]
+        return self.save_to_file(self._PEMS_SEP.join(pems))
+
+
+@IOPlugin.register(path='fullchain.pem', typ=OpenSSL.crypto.FILETYPE_PEM)
+class FullChainFile(FileIOPlugin, OpenSSLIOPlugin):
+    """Full chain file plugin."""
+
+    def persisted(self):
+        return self.Data(account_key=False, key=False, cert=True, chain=True)
+
+    def load_from_content(self, content):
+        pems = [pem for pem in self.split_pems(content)]
+        if len(pems) < 2:
+            raise Error("Not enough PEM encoded messages were found on {0}, "
```

Comment:
Grammar:

1. s/on/in/
2. Use a semicolon, not a comma.

Same thing the other places this message shows up.